manager/state/raft/raft.go

```diff
@@ -1206,10 +1206,63 @@ func (n *Node) applyRemoveNode(cc raftpb.ConfChange) (err error) {
 	// If the node from where the remove is issued is
 	// a follower and the leader steps down, Campaign
 	// to be the leader.
-	if cc.NodeID == n.Leader() {
+
+	members := n.cluster.Members()
+	if cc.NodeID == n.Leader() && !n.IsLeader() {
 		if err = n.Campaign(n.Ctx); err != nil {
 			return err
 		}
+	} else if cc.NodeID == n.Config.ID && n.IsLeader() && len(members) == 2 {
+		// if there are only 2 nodes in the cluster, and leader is leaving
+		// before closing the connection, leader has to ensure that follower gets
+		// noticed about this raft conf change commit. Otherwise, follower would
+		// assume there are still 2 nodes in the cluster and won't get elected
+		// into the leader by acquiring the majority (2 nodes)
+		//
+		// adopt the best-effort strategy to send this notice, because the raft conf change
+		// already gets committed locally, there is no approach to revert the change
+
+		status := n.Node.Status()
+		var conn *membership.Member
+
+		// get peer
+		for id, member := range members {
+			if id == n.Config.ID {
+				continue
+			}
+
+			conn = member
+			break
+		}
```

Comment:
In this case, what kind of action I should apply, return an error or continue the removal?